src/alert.js

```diff
@@ -52,23 +56,35 @@ export default class extends Controller {
 
   close() {
     this.hide()
+
     setTimeout(() => {
       this.element.remove()
-    }, 1100)
-
+    }, this.removeAfter)
   }
 
   show() {
-    this.element.setAttribute(
-      'style',
-      "transition: 1s; transform:translate(0, 0);",
-    )
+    this.element.classList.add(...this.showClasses)
+    this.element.classList.remove(...this.hideClasses)
   }
 
   hide() {
-    this.element.setAttribute(
-      'style',
-      "transition: 1s; transform:translate(400px, 0);",
-    )
+    this.element.classList.add(...this.hideClasses)
+    this.element.classList.remove(...this.showClasses)
+  }
+
+  get removeAfter() {
+    if (this.hasRemoveDelayValue) {
+      return this.removeDelayValue
+    } else {
+      return 1100
+    }
+  }
+
+  get showAfter() {
+    if (this.hasShowDelayValue) {
+      return this.showDelayValue
```

Comment:
Can this be inlined like: `this.showDelayValue || 200`

I can't remember if Stimulus raises an error or just returns undefined for values that aren't set.

Reply:
I wanted to do that, but Stimulus parses unset number values to 0: https://stimulus.hotwired.dev/reference/values#types